service/controller/aws/v14/resource/kubeconfig/current.go

```diff
@@ -0,0 +1,39 @@
+package kubeconfig
+
+import (
+	"context"
+	"fmt"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v14/key"
+)
+
+func (r *StateGetter) GetCurrentState(ctx context.Context, obj interface{}) ([]*corev1.Secret, error) {
+	customObject, err := awskey.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterGuestConfig := awskey.ClusterGuestConfig(customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	secretName := fmt.Sprintf("%s-kubeconfig", clusterGuestConfig.ID)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding kubeconfig secret %#q", secretName))
+
+	secret, err := r.k8sClient.CoreV1().Secrets("").Get(secretName, metav1.GetOptions{})
```

Comment:
```suggestion
	secret, err := r.k8sClient.CoreV1().Secrets(r.resourceNamespace).Get(secretName, metav1.GetOptions{})
```

Doesn't this need to use the setting?

Reply:
I check our CP setting and all namespace was set to default w/o custom namespace, do you think it makes sense to change?

Reply:
I think the secret should be created in the `giantswarm` namespace. The `app-operator-api` cert has a lot of permissions so it should be in a namespace that is clearly ours.

Soon we want to give customers access to CP clusters and they may get access to the `default` namespace.